balancer/internal/wrr/wrr.go

```diff
@@ -0,0 +1,26 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package wrr
+
+// WRR defines an interface that implements weighted round robin.
+type WRR interface {
```

Comment:
Does WRR need to be thread safe? Or if there's a guarantee that Add or Next won't be called concurrently. It's better to document here clearly.

Reply:
Next() needs to be thread safe.

Add() I'm not sure. I'm leaving it undocumented. Will add more requirements when adding new implementations.